### PR TITLE
Explorer: add trace for `InstantiateType`

### PR DESCRIPTION
**Pull request by prabhatexit0**
Adds trace output for `Interpreter::InstantiateType` method.

### CODE REVIEW

File: explorer/interpreter/interpreter.cpp [hunk 1]
```diff
@@ -502,6 +502,11 @@ auto Interpreter::EvalAssociatedConstant(
 auto Interpreter::InstantiateType(Nonnull<const Value*> type,
                                   SourceLocation source_loc)
     -> ErrorOr<Nonnull<const Value*>> {
+  if (trace_stream_->is_enabled()) {
```
Comment (geoffromer) [inline explorer/interpreter/interpreter.cpp:505]: Could we have a test in `testdata/trace` to cover this?

Comment (prabhatexit0) [reply explorer/interpreter/interpreter.cpp:505]: I modified the `testdata/trace/phase_execution.carbon` to include type instantiation. Will this work?

**Review by geoffromer — APPROVED** [review]
Looks good, but it looks like there are some conflicts to resolve.

**Comment by prabhatexit0** [comment]
> Looks good, but it looks like there are some conflicts to resolve.

Resolved conflicts and updated the phase execution test.

**Pull request merged**